extensions/macos-utilities/src/utils.tsx

```diff
@@ -0,0 +1,76 @@
+import fs from "fs";
+import os from "os";
+import path from "path";
+import { execSync } from "child_process";
+
+export const backupLocation = "/tmp/terminal-history-backup.file.txt";
+
+/// Parse the shell history file
+/// - Parameter string: The string to parse
+/// - Returns: An array of commands
+export function parseShellHistory(string: string) {
+  const reBashHistory = /^: \d+:0;/;
+
+  return string
+    .trim()
+    .split("\n")
+    .map((line) => {
+      if (reBashHistory.test(line)) {
+        const lines = line.split(";").slice(1).join(";");
+        // Remove duplicates and return
+        return [...new Set(lines.split(":"))].join(":");
+      }
+
+      // ZSH just places one command on each line
+      return line;
+    });
+}
+
+/// Get the path to the shell history file
+/// - Returns: The path to the shell history file
+export function shellHistoryPath() {
+  const zshHistoryPath = path.join(os.homedir(), ".zsh_history");
+  const bashHistoryPath = path.join(os.homedir(), ".bash_history");
```

Comment:
Not sure if you've considered this, but instead of reading the files and needing explicit support for different shells, could you instead use `useExec` to run the `history` command and parse the output. The `history` command should be available in almost any major shell, and `useExec` can use the user's login shell I believe, which would then respect the configured shell.

Reply:
The `history` command also supports clearing in at least Fish and Bash.

It seems that `history` is implemented on a per-shell basis, so there might be some different support needed here, but using it rather than files would at least reduce the need to support custom configuration locations, and between bash/zsh/fish you'd probably have almost all setups covered.

Reply:
That was my first though as far as parsing out the history command, however, history acts on an non-interactive shell and therefore, I was unable to ever get any results from this approach. My best way to do this was to obtain which shell it was with the `process.env.SHELL` variable and then from there that can determine which file I grab. I will not be able to use the `useExec` command. Unless you've found a way, I wasn't able to.

I took a different approach and looked into the `shell-history` module and that uses the same approach as I have. If you have a solution for this, I am more than willing to adopt it.